### PR TITLE
added main image to item model

### PR DESCRIPTION
**Pull request by bgalek**
added image to Item model

### CODE REVIEW

**Comment by mthenw** [comment]
I will look at it later today

File: test/model/item.js [hunk 1]
```diff
@@ -12,10 +12,13 @@ describe('Item model', function () {
     });
 
     it('should return basic info (id, name, location)', function () {
+        console.log('test');
```
Comment (mthenw) [inline test/model/item.js]: please remove

File: lib/model/item.js [hunk 2]
```diff
@@ -7,13 +7,24 @@ function Item(apiItem, client) {
...
+    this._name = apiItem.itemListInfoExt.itName;
+    this._sellerId = apiItem.itemListInfoExt.itSellerId;
+    this._location = apiItem.itemListInfoExt.itLocation;
+    this._itIsNewUsed = apiItem.itemListInfoExt.itIsNewUsed;
+    this._mainImage = '';
+    if (typeof apiItem.itemImgList !== 'undefined') {
+        if (typeof apiItem.itemImgList[0] !== 'undefined') {
+            if (typeof apiItem.itemImgList[0].item !== 'undefined') {
+                if (typeof apiItem.itemImgList[0].item[2] !== 'undefined') {
+                    this._mainImage = apiItem.itemImgList[0].item[2].imageUrl;
```
Comment (mthenw) [inline lib/model/item.js]: would be much more readable if you would do something like:

```
if (apiItem.itemImgList && apiItem.itemImgList[0] && apiItem.itemImgList[0].item[2]) {
    this._mainImage = apiItem.itemImgList[0].item[2].imageUrl
}
```

File: lib/model/item.js [hunk 3]
```diff
@@ -27,6 +38,12 @@ Item.prototype =  {
         this._client.getUser(this._sellerId, function (err, user) {
             callback(err, user);
         });
+    },
+    get mainImage() {
+        return this._mainImage;
+    },
+    get itIsNewUsed() {
```
Comment (mthenw) [inline lib/model/item.js]: NONONONONO, no IT props. please make two functions isUsed(), and isNew()

File: test/model/item.js [hunk 4]
```diff
@@ -31,8 +34,10 @@ describe('Item model', function () {
         }));
 
         var item = new Item({
-            itId: 1,
-            itSellerId: 2,
+            itemListInfoExt: {
+                itId: 1,
+                itSellerId: 2
+            }
         }, client);
```
Comment (mthenw) [inline test/model/item.js:49]: please add tests about image

File: lib/model/item.js [hunk 5]
```diff
@@ -7,13 +7,24 @@ function Item(apiItem, client) {
...
-    this._id = apiItem.itId;
-    this._name = apiItem.itName;
-    this._sellerId = apiItem.itSellerId;
-    this._location = apiItem.itLocation;
+    this._id = apiItem.itemListInfoExt.itId;
+    this._name = apiItem.itemListInfoExt.itName;
+    this._sellerId = apiItem.itemListInfoExt.itSellerId;
+    this._location = apiItem.itemListInfoExt.itLocation;
+    this._itIsNewUsed = apiItem.itemListInfoExt.itIsNewUsed;
+    this._mainImage = '';
```
Comment (mthenw) [inline lib/model/item.js]: by default it should be null

**Comment by mthenw** [comment]
@bgalek I've added some comments. If you don't have time I can fix it. Just let me know. And thank you for contrib :+1:

**Comment by bgalek** [comment]
already fixing :)

2014-11-27 22:44 GMT+01:00 Maciej Winnicki notifications@github.com:

**Comment by bgalek** [comment]
fixes made, please check now ;)

**Comment by mthenw** [comment]
@bgalek version 1.0.0 released to NPM :)

**Pull request merged**